### ISSUE TITLE
Splits to a random Past iteration instead of moving work items forward

### ISSUE THREAD

**Issue by Arkaine55** [snuziale/vsts-extension-split-work#14]
It is unclear when using the Split tool what iteration incomplete work will be moved to, and after using the tool (TFS 2015 update 3 on prem) we have experienced the work being moved to random/inconsistent past iterations.

Eg: with these 4 example iterations:
Past
- 2017-06-23 Release
- 2017-07-07 Form Rework
Current
- 2017-07-21 SIM Release
Future
- 2017-08-04

"Splitting" PBIs from the 2017-07-21 release we would expect them to goto the 2017-08-04 but some went to 2017-06-23 some went to 2017-07-07.

**Comment by beccamc** [comment]
There are two possible reasons I can think of for this... 
1. Are there dates assigned for all of these sprints?
2. Did you perform the split directly from the form or from the work item hub? 

To determine the "next iteration" this extension pulls all of the iterations chronologically for the "current" team, and finds the next one. Sometimes the order of these iterations is odd if they don't have dates, or if they have overlapping dates. This order is determined by Azure Devops internal system, and you can check the order Azure Devops returns these in by using the [iteration rest api](https://docs.microsoft.com/en-us/rest/api/azure/devops/work/iterations/list?view=azure-devops-rest-5.0). 

You might also run into a problem if you are splitting work items from non-agile pages. We are pulling your iterations based on your team context. With the changes to navigation in 2018, there is no longer a "team context" on every page (e.g. work item hub - this shows all your work items and doesn't have a concept of team).  This means from the Backlog, Board or Sprints hub we know what team you are splitting from and fetch the right iterations. If you open the work item form from somewhere else, say the PR page, we don't know what team is associated with this work item (multiple teams could own an area path). We do a best effort, but we may fetch the settings of a different team. If they don't subscribe to iteration 2017-08-04, we won't be able to move it there. 

tl;dr - Setup your future iteration with dates; Split should work from the Backlogs, Boards and Sprints pages. 

If none of these seem to fix, let me know!
